app/controllers/application_controller.rb

```diff
@@ -282,6 +282,15 @@ def redirect_to(*args)
     end
   end
 
+  # Objects that belong to a single observation:
+  def redirect_to_observation_or_object(obj)
```

Comment:
I know we talked about this earlier, so I want to focus on only the most important part -- is there a way we can better expose that this method relies on @back_object?

I see it is used 7 times ... 7? why an odd number? strange! ... in herbarium_records_controller and collection_number_controller.  That's not that many.  How about explicitly passing @back_object in?

Reply:
Yeah it's kind of a switch, isn't it? It's a weird method.

Should it be called this and have a safety back to "/"
```ruby
redirect_to_back_object_or_object(back_obj, obj)
  if back_obj
    redirect_with_query(back_obj.show_link_args)
  elsif obj
    redirect_with_query(obj.index_link_args)
  else
    redirect_with_query("/")
  end
end
```

Reply:
Seven times because there's an extra check for Herbarium permission in the HR controller

Reply:
Yes, that works so much better for me, thank you!